src/common-util/src/main/java/ee/ria/xroad/common/Version.java

```diff
@@ -78,6 +82,48 @@
         }
     }
 
+    /**
+     * Outputs version information and a warning if JVM version is not in the given range
+     * @param minJavaVersion the minimum supported version
+     * @param maxJavaVersion the maximum supported version
+     */
+    public static void outputVersionInfo(String appName, int minJavaVersion, int maxJavaVersion) {
+        // print app name + version and java vendor name + runtime version
+        String vendorString = System.getProperty(JAVA_VENDOR_PROPERTY);
+        String versionString = System.getProperty(JAVA_VERSION_PROPERTY);
+        String runtimeString = System.getProperty(JAVA_RUNTIME_PROPERTY);
+        StringBuilder vendorVersionBuilder = new StringBuilder();
+        if (vendorString != null) {
+            vendorVersionBuilder.append(vendorString + " ");
+        }
+        vendorVersionBuilder.append(runtimeString);
+        System.out.println(
+                String.format("%s %s (Java %s)", appName, XROAD_VERSION, vendorVersionBuilder.toString()));
+        // java.version system property exists in every JVM
+        if (versionString.startsWith("1.")) {
+            // Java 8 or lower has format: 1.6.0_23, 1.7.0, 1.7.0_80, 1.8.0_211
+            versionString = versionString.substring(2, VERSION_STRING_SUFFIX_LENGTH);
+        } else {
+            // Java 9 or higher has format: 9.0.1, 11.0.4, 12, 12.0.1
+            int dot = versionString.indexOf(".");
+            if (dot != -1) {
+                versionString = versionString.substring(0, dot);
+            }
+        }
+        int result = Integer.parseInt(versionString);
```

Comment:
This could fail with numberformatexception.

Reply:
Added exception handling